lib/optimization/table_data.go

```diff
@@ -208,7 +208,7 @@ func (t *TableData) ToFqName(kind constants.DestinationKind, escape bool, upperc
 		}))
 	case constants.BigQuery:
 		// The fully qualified name for BigQuery is: project_id.dataset.tableName.
-		return fmt.Sprintf("%s.%s.%s", opts.BigQueryProjectID, t.TopicConfig.Database, t.Name(uppercaseEscNames, &sql.NameArgs{
+		return fmt.Sprintf("`%s`.%s.%s", opts.BigQueryProjectID, t.TopicConfig.Database, t.Name(uppercaseEscNames, &sql.NameArgs{
```

Comment:
Should we escape `t.TopicConfig.Database` too?

Reply:
Yeah we should

Reply:
This is wrong lol: https://github.com/googleapis/google-cloud-go/blob/1ef5b1917bb9a1271c3fb152413ec0e74163164d/bigquery/table.go#L684C27-L684C29

It is what we originally had

Reply:
Hah yeah I wondered about that.